pkg/manager/internal/controllers/organizationroletemplate_controller.go

```diff
@@ -70,9 +110,29 @@ func (r *OrganizationRoleTemplateReconciler) Reconcile(req ctrl.Request) (ctrl.R
 }
 
 func (r *OrganizationRoleTemplateReconciler) SetupWithManager(mgr ctrl.Manager) error {
-
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&corev1alpha1.OrganizationRoleTemplate{}).
+		Watches(&source.Kind{Type: &corev1alpha1.Organization{}}, &handler.EnqueueRequestsFromMapFunc{
+			ToRequests: handler.ToRequestsFunc(func(mapObject handler.MapObject) (out []ctrl.Request) {
+				organization := mapObject.Object.(*corev1alpha1.Organization)
+				if !organization.IsReady() {
+					return
+				}
+				templates := &corev1alpha1.OrganizationRoleTemplateList{}
+				if err := r.Client.List(context.Background(), templates); err != nil {
+					// This will makes the manager crashes, and it will restart and reconcile all objects again.
```

Comment:
```suggestion
					// This will makes the manager crashes, and it will restart and reconcile all objects again.
				        utilruntime.HandleError(err)
```

Reply:
Call the necessary error handlers before panicking.

Reply:
I remembered that is not needed: for example, controller-runtime controller pkg calls this: https://github.com/kubernetes-sigs/controller-runtime/blob/bb96678bedc9d98361be219cb21562eb99ab9e15/pkg/internal/controller/controller.go#L133
And HandleCrash will crashes after logging the panic message:https://github.com/kubernetes/apimachinery/blob/681a08151eac875afc5286670195105118d3485d/pkg/util/runtime/runtime.go#L45

Reply:
ok